rcl_yaml_param_parser/src/parse.c

```diff
@@ -585,6 +594,166 @@ _validate_name(const char * name, rcutils_allocator_t allocator)
   return ret;
 }
 
+rcutils_ret_t
+_validate_tag(const char * tag, uint32_t line_num)
+{
+  if (tag == NULL) {
+    return RCUTILS_RET_ERROR;
+  }
+
+  if ((0 == strcmp(tag, YAML_NULL_TAG)) ||
+    (0 == strcmp(tag, YAML_BOOL_TAG)) ||
+    (0 == strcmp(tag, YAML_STR_TAG)) ||
+    (0 == strcmp(tag, YAML_INT_TAG)) ||
+    (0 == strcmp(tag, YAML_FLOAT_TAG)) ||
+    (0 == strcmp(tag, YAML_TIMESTAMP_TAG)) ||
+    (0 == strcmp(tag, YAML_SEQ_TAG)) ||
+    (0 == strcmp(tag, YAML_MAP_TAG)))
+  {
+    return RCUTILS_RET_OK;
+  }
+
+  RCUTILS_SET_ERROR_MSG_WITH_FORMAT_STRING(
+    "Unsupported tag %s at line %d", (char *)tag, line_num);
+  return RCUTILS_RET_ERROR;
+}
+
+rcutils_ret_t
+_validate_bool_value(
+  const char * const value,
+  data_types_t * val_type,
+  void ** ret_val,
+  const rcutils_allocator_t allocator)
+{
+  rcutils_ret_t ret = RCUTILS_RET_ERROR;
+
+  if ((0 == strcmp(value, "Y")) ||
+    (0 == strcmp(value, "y")) ||
+    (0 == strcmp(value, "yes")) ||
+    (0 == strcmp(value, "Yes")) ||
+    (0 == strcmp(value, "YES")) ||
+    (0 == strcmp(value, "true")) ||
+    (0 == strcmp(value, "True")) ||
+    (0 == strcmp(value, "TRUE")) ||
+    (0 == strcmp(value, "on")) ||
+    (0 == strcmp(value, "On")) ||
+    (0 == strcmp(value, "ON")))
+  {
+    *val_type = DATA_TYPE_BOOL;
+    *ret_val = allocator.zero_allocate(1U, sizeof(bool), allocator.state);
+    if (NULL != *ret_val) {
+      *((bool *)*ret_val) = true;
+    }
+    ret = RCUTILS_RET_OK;
+    goto final;
```

Comment:
`goto`  is unnecessary.  Directly do return.

Reply:
Updated.